### PR TITLE
Check that Azure blob md5 field is present

### PR DESCRIPTION
**Pull request by pdostal**
This continues the effort of b5cea54643aada0d0ca134e3a6c044a9da840ea7 (#16770).

- Related ticket: [poo#126914](https://progress.opensuse.org/issues/126914)
- Verification run: [15-SP5-Azure-BYOS](https://pdostal-server.suse.cz/t4263), [15-SP5-EC2-ARM](https://pdostal-server.suse.cz/t4264)

### CODE REVIEW

File: lib/publiccloud/azure.pm [hunk 1]
```diff
@@ -58,6 +58,7 @@ sub get_image_id {
     return $self->SUPER::get_image_id($img_url);
 }
 
+# asdf
```
Comment (asmorodskyi) [inline lib/publiccloud/azure.pm:61]: leftover ?

File: lib/publiccloud/azure.pm [hunk 2]
```diff
@@ -67,8 +68,29 @@ sub find_img {
...
+    record_info('BLOB INFO', script_output($cmd_show));    # This is temporal debug
+    my $start_time = time();
+    my $upload_time = 600;
+    my $completed = 0;
+    while (time() - $start_time < $upload_time) {
+        my $blob_info = decode_azure_json(script_output($cmd_show));
+        record_info('MD5', "Blob MD5: $blob_info->{md5}");
+        # Check that MD5 metadata exists and is not empty
+        # We don't compare the value yet, because we don't always have the downloaded.
+        $completed = 1 if ($blob_info->{md5} && length($blob_info->{md5}) > 0);
```
Comment (mpagot) [inline lib/publiccloud/azure.pm:87]:
```
length($blob_info->{md5}) > 0
```

formally it is never 0, even if blob does not have the MD5, the `az storage blob show` return you the string `null` or something similar.

**Comment by asmorodskyi** [comment]
generally I like it but I have doubts about this waiting loop , I have a feeling that we solving problem in wrong place . Normal there should be single job which upload image and all others should not start before it is done and we have openQA which make sure that it works like that . Let's check why we having several upload_images uploading same image in parallel ( let's discuss this in the call )

File: lib/publiccloud/azure.pm (hunk 2 above)
Comment (pdostal) [reply lib/publiccloud/azure.pm]: Pity that you did not raise this concern yesterday.

I cannot compare it with the md5 hash of the file. See the comment above.

The md5 hash should have always 32 characters, so please check now.

**Review by mpagot — APPROVED** [review]
LGTM

**Review by ricardobranco777 — APPROVED** [review]
LGTM

**Pull request merged**